himan-scripts/CB-TCU-cloud.lua

```diff
@@ -64,10 +74,11 @@ for i=1, #EL500 do
   res[i] = Missing
 
   --TCU
-  if ((Tbase[i]-Ttop[i]>TCUlimit) and (NL[i]>0) and (CIN500[i]>CINlimTCU)) then
-    res[i] = FlightLevel_(EL500[i]*100)
+  if (EL500[i] - LCL500[i] > TCUlimit) then
+    --we don't use vertical search for flight level of EL500 but calculate directly from EL500 pressure
+    res[i] = FlightLevel_(pEL500[i] * 100)
     --Limit top value
-    if (CAPE500[i]>math.exp(1)) then
+    if (CAPE500[i] > math.exp(1)) then
       --Add for overshooting top based on CAPE 
       res[i] = -(res[i] + CAPE500[i]  / (math.log(CAPE500[i]) * 10))
```

Comment:
In smarttool the last term (10) has a division operator, not multiplication. I tried to think if you are doing some kind of mathematical trick here but I couldn't get it :D

Reply:
I put () around it which is the mathematical trick you missed. On a high nerd level I think multiplication is a cheaper operation than division.

exp(1) is used to generate the constant e that LUA doesn't provide. This was needed because we had some problems where CAPE was small and then the logarithm function goes to 0 for ln(1) and division by 0 is bad as well as for values below 1 it gives negative numbers that potentially turns the sign or the result around. Which by the logic of this macro turns CB into TCU. Probably it's not wise to have two things in one parameter and we should output CB and TCU as two separate fields but that's what it is now.

Reply:
I was just wondering why e(1), does it have some special significance. Why doesn't he just declare "const CAPElimit = 2.78".

I agree it's a bit weird that this parameter has different interpretation depending on the sign of the data value.

Reply:
Yeah, I think I'll change it to that CAPElimit version then.